Mathlib/Data/Countable/Basic.lean

```diff
@@ -84,6 +84,18 @@ instance [Nonempty α] [Uncountable β] : Uncountable (α × β) := by
   inhabit α
   exact (Prod.mk.inj_left default).uncountable
 
+lemma countable_left_of_prod_of_nonempty [Nonempty β] (h : Countable (α × β)) : Countable α := by
+  contrapose h
+  rw [not_countable_iff] at *
+  infer_instance
+
+lemma countable_right_of_prod_of_nonempty [Nonempty α] (h : Countable (α × β)) : Countable β := by
+  contrapose h
+  rw [not_countable_iff] at *
+  infer_instance
+
+instance [Countable (α × β)] : Countable (β × α) := Countable.of_equiv _ (Equiv.prodComm α β)
```

Comment:
Do you need this instance for an application? Instances should not be added lightly, as they are one of the main performance bottlenecks in Mathlib. Otherwise, maybe make it a lemma?

Reply:
Let's make it a lemma, yeah